### DISCUSSION TITLE
外部密钥的生成权限问题

### DISCUSSION THREAD

**Discussion by colorthoro** [dashingsoft/pyarmor#1875] [Q&A]
很久以前，谁都可以生成外部密钥，可以在密钥里添加自定义的密码，然后在项目中添加辅助认证代码检查密钥是否匹配。
现在虽然也看到外部密钥生成时可以添加自定义数据，但“打包使用外部密钥加密的脚本实践”里没有提到自定义密码的设置了，是不需要了吗？那么是什么原理保证只有自己生成的外部密钥才可用呢？

**Comment by colorthoro** [comment]
现在还是同样的原理吧？好像是用加密脚本来保证，并且不用在项目代码注释锚点了，对吗？

**Comment by jondy** [comment]
外部密钥的数据怎么使用完全是加密脚本自己决定的
